receiver/receiver.go

```diff
@@ -195,3 +197,77 @@ func MakeFactoryMap(factories ...Factory) (map[component.Type]Factory, error) {
 	}
 	return fMap, nil
 }
+
+// Builder receiver is a helper struct that given a set of Configs and Factories helps with creating receivers.
+type Builder struct {
+	cfgs      map[component.ID]component.Config
+	factories map[component.Type]Factory
```

Comment:
Should disallow implementing the method to leave us room for improvement given that we have the `NewBuilder`? Otherwise, NewBuilder may be redundant?